diff --git a/me/jet315/repairtokens/utils/ReflectionUtils.java b/me/jet315/repairtokens/utils/ReflectionUtils.java
index 0a06675..f495bbd 100644
--- a/me/jet315/repairtokens/utils/ReflectionUtils.java
+++ b/me/jet315/repairtokens/utils/ReflectionUtils.java
@@ -84,8 +84,10 @@ public class ReflectionUtils {
                }
             }
          } else {
-            nmsVersion = "v1_20_R4";
-            serverVersion = "v1_20_R4";
+            // Luma start - 1.21.11 support
+            nmsVersion = "";
+            serverVersion = "";
+            // Luma end
          }
 
          fullServerVersion = Bukkit.getVersion();
